### DISCUSSION TITLE
Slow application build time and suspicious internal IgnorePlugin

### DISCUSSION THREAD

**Discussion by inside** [vercel/next.js#23114] [Help]
Hello :wave: 

I noticed [the IgnorePlugin used in the next's webpack config](https://github.com/vercel/next.js/blob/b80fdfb8287980c8f9ae3c9c591ab796ad00baf1/packages/next/build/webpack-config.ts#L1120) spends more than 3 minutes in our huge app's build process. It's purpose is to [avoid bundling (under 1k gzipped) the `react-is` module when in production mode](https://github.com/vercel/next.js/blob/2d207fb11d16c02d6d7f419f4001a63dcbe7c944/packages/next/next-server/lib/router/router.ts#L1335). I guess the more files you have in your app, the more time it will take to build.

Other webpack plugins I could find (ContextExclusionPlugin, ContextReplacementPlugin, NormalModuleReplacementPlugin) are not a fit for that use case and passing the `checkResource` filter function to IgnorePlugin (when I manually edit next's webpack config) gives me about the same ~3 minutes build time.

What are your thoughts about decreasing the build time in this case?

Please find below the output of the `speed-measure-webpack-plugin` configured in our app:

```
 SMP  ⏱
General output time took 12 mins, 38.32 secs

 SMP  ⏱  Plugins
IgnorePlugin took 3 mins, 23.68 secs
TerserPlugin took 39.89 secs
StatsWriterPlugin took 15.64 secs
NextJsSsrImportPlugin took 12.27 secs
CssMinimizerPlugin took 7.95 secs
HashedModuleIdsPlugin took 2.35 secs
DropClientPage took 1.45 secs
WellKnownErrorsPlugin took 1.44 secs
ChunkNamesPlugin took 1.32 secs
DefinePlugin took 1.16 secs
ContextReplacementPlugin took 0.943 secs
NextMiniCssExtractPlugin took 0.402 secs
BuildManifestPlugin took 0.013 secs
ReactLoadablePlugin took 0.008 secs
PagesManifestPlugin took 0.002 secs
```

**Comment by zyzski** [comment]
I was looking for a way to get more info on build timing so thank you for mentioning that plugin

**Reply by inside** [reply]
You're welcome :+1:

**Comment by ramon-villain** [comment]
@inside Did you try using yarn patch or patch-package? That was the only way I found to get the plugin removed
